src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -201,49 +201,46 @@ private void ProcessEvent(WorkspaceChangeEventArgs args, string eventName)
                         EnqueueFullSolutionEvent(args.NewSolution, InvocationReasons.DocumentAdded, eventName);
                         break;
 
-                    case WorkspaceChangeKind.SolutionChanged:
-                    case WorkspaceChangeKind.SolutionReloaded:
-                        EnqueueSolutionChangedEvent(args.OldSolution, args.NewSolution, eventName);
-                        break;
-
                     case WorkspaceChangeKind.SolutionRemoved:
+                    case WorkspaceChangeKind.SolutionCleared:
                         EnqueueFullSolutionEvent(args.OldSolution, InvocationReasons.SolutionRemoved, eventName);
                         break;
 
-                    case WorkspaceChangeKind.SolutionCleared:
-                        EnqueueFullSolutionEvent(args.OldSolution, InvocationReasons.SolutionRemoved, eventName);
+                    case WorkspaceChangeKind.SolutionChanged:
+                    case WorkspaceChangeKind.SolutionReloaded:
+                        EnqueueSolutionChangedEvent(args.OldSolution, args.NewSolution, eventName);
                         break;
```

Comment:
moved consistent to have `Add` tehn `Remove` then `Changed/Reloaded` for all types.